### DISCUSSION TITLE
Run ege function api call as custom user role

### DISCUSSION THREAD

**Discussion by SebasScript** [supabase/supabase#20471] [Questions]
I can create a supabase client in an edge function with the service role key  to run queries under the service role.
Is it possible to create custom postgress roles and do the same? Intialise the client under the custom user role to than run queries with that role? (without using a custom user that is loged in with a password)

**Comment by GaryAustin1** [comment]
You can mint a jwt with your role claim and use that in the Authorization header (option in createClient to set it).  I'm not sure you can change out the apikey role, but I've never tried it.   Either way you mint a jwt using the jwt secret.  https://supabase.com/docs/learn/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase
